tests/test_core.py

```diff
@@ -57,12 +59,78 @@ class test_Variable(unittest.TestCase):
 
     def test_init(self):
         """Test construction with different bounds."""
-        raise NotImplementedError()
+        v = relentless.core.Variable(value=1.0, const=True)
+        self.assertAlmostEqual(v.value, 1.0)
+        self.assertEqual(v.const, True)
+
+        v = relentless.core.Variable(value=1.2, low=0.0, high=2.0)
+        self.assertAlmostEqual(v.value, 1.2)
+        self.assertAlmostEqual(v.low, 0.0)
+        self.assertAlmostEqual(v.high, 2.0)
+
+        v = relentless.core.Variable(value=-1.2, low=0.5)
+        self.assertAlmostEqual(v.value, 0.5)
+        self.assertAlmostEqual(v.low, 0.5)
+
+        v = relentless.core.Variable(value=2.2, high=2.0)
+        self.assertAlmostEqual(v.value, 2.0)
+        self.assertAlmostEqual(v.high, 2.0)
+
+        with self.assertRaises(ValueError):
+            v = relentless.core.Variable(value=4)
```

Comment:
This test is totally correct but indicates bad design on my part. Even though we think of these as being floats, the user will sometimes type an int (e.g., `4`) instead of a float (`4.0`), and we should allow both. Let's fix it in the Variable class and convert this to a test that passes using an int.

We still want to restrict the types of the variables to those we know how to work with, so in the `isinstance` call inside `value`, check `(float,int)` instead of just `float`. This should cover the two main data types. Please also update the documentation for `__init__` to allow float or int.

You can still keep a test that checks if this raises, but use some silly thing like a `str` and/or a tuple.